tests/test_skills/test_apy_estimation_abci/test_apis.py

```diff
@@ -97,7 +100,43 @@ def test_eth_price(spooky_specs: SpecsType, eth_price_usd_q: str) -> None:
         res = make_request(api.get_spec(), eth_price_usd_q)
         eth_price = ast.literal_eval(api.process_response(DummyMessage(res.content))[0]["ethPrice"])  # type: ignore
 
-        assert isinstance(eth_price, float)
+        assert eth_price == 0.4183383786296383
+
+    @staticmethod
+    def test_eth_price_non_indexed_block(
+        spooky_specs: SpecsType, eth_price_usd_q: str
+    ) -> None:
+        """Test SpookySwap's eth price request from subgraph, when the requesting block has not been indexed yet."""
+        spooky_specs["response_key"] = "errors"
+        api = SpookySwapSubgraph(**spooky_specs)
+
+        # replace the block number with a huge one, so that we get a not indexed error
+        current_number = "3830367"
+        largest_acceptable_number = "2147483647"
+        eth_price_usd_q = eth_price_usd_q.replace(
+            current_number, largest_acceptable_number
+        )
+        expected_error = (
+            r"Unknown error encountered!\nRaw response: \{'errors': \[\{'message': 'Failed to decode `block.number` "
+            "value: `subgraph QmPJbGjktGa7c4UYWXvDRajPxpuJBSZxeQK5siNT3VpthP has only indexed up to block number "
+            "\\d+ and data for block number 2147483647 is therefore not yet available`'}]}"
+        )
+        with pytest.raises(ValueError, match=expected_error):
+            make_request(api.get_spec(), eth_price_usd_q)
+
+    @staticmethod
+    def test_regex_for_indexed_block_capture() -> None:
+        """Test the regex for capturing the indexed block."""
+        error_message = (
+            "Failed to decode `block.number` value: `subgraph QmPJbGjktGa7c4UYWXvDRajPxpuJBSZxeQK5siNT3VpthP has only "
+            "indexed up to block number 3730367 and data for block number 3830367 is therefore not yet available`"
+        )
+        match = re.match(NON_INDEXED_BLOCK_RE, error_message)
+        assert match is not None
+        assert match.groups() == ("3730367",)
+
+        error_message = "new message 3730367"
+        assert re.match(NON_INDEXED_BLOCK_RE, error_message) is None
```

Comment:
can we also have a test that checks (in several attempts) to verify that the regex works on the current data we're retrieving from spookyswap? To ensure that we don't end up in a situation where they change their pattern without us noticing immediately

Reply:
This is what the above method does. I will refactor it to use the `NON_INDEXED_BLOCK_RE`.